admin/cron/hourly.sh

```diff
@@ -24,6 +24,81 @@ fi
 
 . ./admin/functions.sh
 
+echo "MBS-10921 (1/3): Drop wrongly-listed edits in artist history..."
+
+./admin/psql READWRITE <<EOSQL
+SET LOCAL statement_timeout = 0;
+DELETE FROM edit_artist
+WHERE (edit, artist) IN (
+	WITH recording_artist AS (
+		SELECT r.id recording, acn.artist
+		FROM recording r
+		JOIN artist_credit ac ON r.artist_credit = ac.id
+		JOIN artist_credit_name acn ON ac.id = acn.artist_credit
+	)
+	SELECT ea.edit, ea.artist
+	FROM edit_artist ea
+	JOIN artist a ON ea.artist = a.id
+	JOIN edit e ON ea.edit = e.id
+	JOIN edit_data ed ON e.id = ed.edit
+	JOIN recording r ON jsonb_extract_path_text(ed.data, 'entity0', 'id')::int = r.id
+	WHERE (e.type >= 90 AND e.type <= 99)
+	AND (jsonb_extract_path_text(ed.data, 'type0') = 'recording' AND jsonb_extract_path_text(ed.data, 'type1') = 'work')
+	AND ea.artist NOT IN (SELECT artist FROM recording_artist ra WHERE r.id = ra.recording)
+	AND e.id >= 70658947 AND e.id < 71006471
+	LIMIT 4000
+)
+RETURNING *;
+EOSQL
+
+echo "MBS-10921 (2/3): Drop wrongly-listed edits in recording history..."
+
+./admin/psql READWRITE <<EOSQL
+SET LOCAL statement_timeout = 0;
+DELETE FROM edit_recording
+WHERE (edit, recording) IN (
+	SELECT er.edit, er.recording
+	FROM edit_recording er
+	JOIN recording r ON er.recording = r.id
+	JOIN edit e ON er.edit = e.id
+	JOIN edit_data ed ON e.id = ed.edit
+	JOIN recording r2 ON jsonb_extract_path_text(ed.data, 'entity0', 'id')::int = r2.id
+	WHERE (e.type >= 90 AND e.type <= 99)
+	AND (jsonb_extract_path_text(ed.data, 'type0') = 'recording' AND jsonb_extract_path_text(ed.data, 'type1') = 'work')
+	AND r.id != r2.id
+	AND e.id >= 70658947 AND e.id < 71006471
+	LIMIT 8000
+)
+RETURNING *;
+EOSQL
+
+echo "MBS-10921 (3/3): Drop wrongly-listed edits in release history..."
+
+./admin/psql READWRITE <<EOSQL
+SET LOCAL statement_timeout = 0;
+DELETE FROM edit_release
+WHERE (edit, release) IN (
+	WITH recording_release AS (
+		SELECT recording.id AS recording, release.id AS release
+		FROM release
+		JOIN medium ON release.id = medium.release
+		JOIN track ON medium.id = track.medium
+		JOIN recording ON track.recording = recording.id
+	)
+	SELECT er.edit, er.release
+	FROM edit_release er
+	JOIN edit e ON er.edit = e.id
+	JOIN edit_data ed ON e.id = ed.edit
+	JOIN recording r ON jsonb_extract_path_text(ed.data, 'entity0', 'id')::int = r.id
+	WHERE (e.type >= 90 AND e.type <= 99)
+	AND (jsonb_extract_path_text(ed.data, 'type0') = 'recording' AND jsonb_extract_path_text(ed.data, 'type1') = 'work')
+	AND r.id NOT IN (SELECT recording FROM recording_release rr WHERE er.release = rr.release)
```

Comment:
PG seems to be having trouble optimizing the `recording_release` CTE, even though there's only a single use of it. This is much faster and *might* be equivalent:
```suggestion
      AND er.release NOT IN (SELECT release FROM medium m JOIN track t ON t.medium = m.id WHERE t.recording = r.id)
```

Reply:
Reworked in commit https://github.com/metabrainz/musicbrainz-server/pull/1581/commits/353ab4192d85de05b3cb9228b63a84e8abe13117 to get rid of CTEs.